perception/traffic_light_ssd_fine_detector/src/nodelet.cpp

```diff
@@ -78,11 +80,17 @@ TrafficLightSSDFineDetectorNodelet::TrafficLightSSDFineDetectorNodelet(
     net_ptr_.reset(new ssd::Net(onnx_file, mode, max_batch_size));
     net_ptr_->save(engine_path);
   }
-  channel_ = net_ptr_->getInputSize()[0];
-  width_ = net_ptr_->getInputSize()[1];
-  height_ = net_ptr_->getInputSize()[2];
-  detection_per_class_ = net_ptr_->getOutputScoreSize()[0];
-  class_num_ = net_ptr_->getOutputScoreSize()[1];
+
+  input_shape_ = net_ptr_->getInputShape();
+  if (is_box_first_) {
+    box_dims_ = net_ptr_->getOutputDimensions(1);
+    score_dims_ = net_ptr_->getOutputDimensions(2);
+  } else {
+    score_dims_ = net_ptr_->getOutputDimensions(1);
+    box_dims_ = net_ptr_->getOutputDimensions(2);
+  }
```

Comment:
```suggestion
    box_dims_ = net_ptr_->getTensorShape("box");
    score_dims_ = net_ptr_->getTensoorShape("score");
```
How about using `getTensorShape`?
https://docs.nvidia.com/deeplearning/tensorrt/api/c_api/classnvinfer1_1_1_i_execution_context.html#a3e2e742086ce307bc03b86122924c57f

Reply:
@wep21 It seems `getTensorShape` is supported from `TensorRT 8.5.1`, but autoware default version is `8.4.2`. So, it would occur build error?

Reply:
@ktro2828 I think this is a solution.
https://github.com/autowarefoundation/autoware.universe/blob/c41ba0638f668bb25053f7f4e61ce0602835e582/perception/traffic_light_classifier/utils/trt_common.cpp#L80-L87

Reply:
@wep21 Thank you for your suggestion! I updated the method  to get tensor shape by its name in https://github.com/autowarefoundation/autoware.universe/pull/3485/commits/3c514707d9f6333ce859117e2e7dd2391f548806
Note that, since the name of tensor `input/boxes/scores` are same in both pytorch-ssd and mmdetection I use it directly.